### PR TITLE
[USM] tests https prefetch lib.so 

### PR DESCRIPTION
**Pull request by nplanel**
### What does this PR do?

Fix bug prefetch bug introduce by a previous [PR](https://github.com/DataDog/datadog-agent/pull/16650)
As we relay only by checking the pid (we don't scan ourself)

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by brycekahle — APPROVED** [review]

**Review by guyarb — APPROVED** [review]
Great catch

File: pkg/network/tracer/testutil/prefetch_file/prefetch_file.go [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+	"fmt"
+	"os"
+	"strconv"
+	"time"
+)
+
+func main() {
+	if len(os.Args) != 3 {
+		fmt.Println("usage: prefetch_file <filename> <wait_time_in_seconds>")
+		return
```
Comment (hmahmood) [inline pkg/network/tracer/testutil/prefetch_file/prefetch_file.go:18]: Would be better to return an non zero error code here (and other locations below) with `os.Exit`

File: pkg/network/tracer/tracer_usm_linux_test.go [hunk 2]
```diff
@@ -211,6 +212,39 @@ func TestHTTPSViaLibraryIntegration(t *testing.T) {
...
+	if _, err = os.Stat(binary); err == nil {
+		return binary
+	}
+
+	srcDir := fmt.Sprintf("%s/testutil/%s", cur, srcPath)
+
+	c := exec.Command("go", "build", "-buildvcs=false", "-a", "-ldflags=-extldflags '-static'", "-o", binary, srcDir)
+	out, err := c.CombinedOutput()
+	t.Log(c, string(out))
+	require.NoError(t, err, "could not build  test binary: %s\noutput: %s", err, string(out))
```
Comment (hmahmood) [inline pkg/network/tracer/tracer_usm_linux_test.go:236]: Extra space after "build "

File: pkg/network/tracer/tracer_usm_linux_test.go [hunk 3]
```diff
@@ -211,6 +212,39 @@ func TestHTTPSViaLibraryIntegration(t *testing.T) {
...
+
+	return binary
+}
+
+func prefetchLib(t *testing.T, filename string) *exec.Cmd {
+	prefetchBin := buildPrefetchFileBin(t)
+	cmd := exec.Command(prefetchBin, filename, "5")
```
Comment (hmahmood) [inline pkg/network/tracer/tracer_usm_linux_test.go:243]: Wouldn't this be treated as 5 nanoseconds by `time.Duration`? You should be able to use `5s` here as that should be parsed correctly by `time.Duration`.

**Comment by guyarb** [comment]
Addressed @hmahmood comments as this PR is urgent

**Review by hmahmood — APPROVED** [review]

**Pull request merged**